src/driver/postgres/PostgresDriver.ts

```diff
@@ -586,14 +586,24 @@ export class PostgresDriver implements Driver {
 
         } else if (columnMetadata.type === "enum" || columnMetadata.type === "simple-enum" ) {
             if (columnMetadata.isArray) {
+                if (value === "{}") return [];
+
                 // manually convert enum array to array of values (pg does not support, see https://github.com/brianc/node-pg-types/issues/56)
-                value = value !== "{}" ? (value as string).substr(1, (value as string).length - 2).split(",") : [];
-                // convert to number if that exists in poosible enum options
+                value = (value as string).substr(1, (value as string).length - 2).split(",").map(val => {
+                    // replace double quotes from the beginning and from the end
+                    val = val.replace(/(^")|("$)/g, "")
+                    // replace double escaped backslash to single escaped e.g. \\\\ -> \\
+                    val = val.replace(/(\\\\)/g, "\\")
+                    // replace escaped double quotes to non-escaped e.g. \"asd\" -> "asd"
+                    return val.replace(/(\\")/g, '"')
```

Comment:
Could be faster? Probably slightly easier to read. Safer for the start/end quotes because it won't fix a broken `"string` with missing start/end quote.

```suggestion
                    // replace double quotes from the beginning and from the end
                    if (val.startsWith(`"`) && val.endsWith(`"`)) val = val.slice(1, -1);
                    // replace double escaped backslash to single escaped e.g. \\ -> \
                    val = val.replaceAll("\\\\", "\\");
                    // replace escaped double quotes to non-escaped e.g. \"asd\" -> "asd"
                    return val.replaceAll(`\\"`, `"`);
```